### PR TITLE
My new christmas cards, with the issues fixed

### PR DESCRIPTION
**Pull request by alicetouhou**

### CODE REVIEW

File: client/src/app/game/toolbar/toolbar.tsx [hunk 1]
```diff
@@ -33,7 +46,7 @@ function Toolbar() {
 			</button>
 
 			{/* Toggle Chat */}
-			{settings.disableChat === 'off' && <ChatItem />}
+			<ChatItem />
```
Comment (Tyrannicodin) [inline client/src/app/game/toolbar/toolbar.tsx]: Why is this changed?

File: common/ailments/museum-collection.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
-import Ailment from './ailment'
-import {GameModel} from '../models/game-model'
-import {CardPosModel} from '../models/card-pos-model'
-import {removeAilment} from '../utils/board'
-import {AilmentT} from '../types/game-state'
-
-class MuseumCollectionAilment extends Ailment {
-	constructor() {
+import Ailment from "./ailment"
```
Comment (Tyrannicodin) [inline common/ailments/museum-collection.ts:1]: Pretty sure the formatting is wrong here

File: common/ailments/protected.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
-import Ailment from './ailment'
-import {GameModel} from '../models/game-model'
-import {CardPosModel, getBasicCardPos, getCardPos} from '../models/card-pos-model'
-import {removeAilment} from '../utils/board'
-import {AilmentT} from '../types/game-state'
-import {isTargetingPos} from '../utils/attacks'
+import Ailment from "./ailment"
```
Comment (Tyrannicodin) [inline common/ailments/protected.ts:1]: Same problem with formatting

File: common/cards/effects/brewing-stand.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ class BrewingStandEffectCard extends EffectCard {
 			name: 'Brewing stand',
 			rarity: 'rare',
 			description:
-				'Attach to a Hermit, every 2 turns consumes 1 item card and heals the Hermit it is attached to 50hp',
+				'Attach to a hermit, every 2 turns, consume 1 item card and heal the hermit it is attached to 50hp',
```
Comment (Tyrannicodin) [inline common/cards/effects/brewing-stand.ts]: Shouldn't Hermit be capital?

File: common/cards/hermits/bigbst4tz2-rare.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+			const coinFlip = flipCoin(opponentPlayer, this.id)
+			if (coinFlip[0] === 'tails') return
+
+			const backlashAttack = new AttackModel({
+				id: this.getInstanceKey(instance, 'backlash'),
+				attacker: attack.target,
+				target: attack.attacker,
+				type: 'secondary',
+				isBacklash: true,
+			}).addDamage(this.id, opponentActiveRow.health)
```
Comment (Tyrannicodin) [inline common/cards/hermits/bigbst4tz2-rare.ts:61]: This might cause problems with stuff like armor

File: common/cards/hermits/katherineeliz-rare.ts [hunk 6]
```diff
@@ -75,7 +74,10 @@ class KatherineelizRareHermitCard extends HermitCard {
 					pickedPlayer.board.rows[rowIndex].effectCard = cardInfo
 
 					return 'SUCCESS'
-				}
+				},
+				onTimeout() {
```
Comment (Tyrannicodin) [inline common/cards/hermits/katherineeliz-rare.ts]: You don't need an empty timeout function

File: common/cards/hermits/pixlriffs-rare.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+			delete player.custom[instance]
+		})
+	}
+
+	public override onDetach(game: GameModel, instance: string, pos: CardPosModel): void {
+		const {player} = pos
+		const instanceKey = this.getInstanceKey(instance)
+
+		player.hooks.onAttack.remove(instance)
+		delete player.custom[instanceKey]
```
Comment (Tyrannicodin) [inline common/cards/hermits/pixlriffs-rare.ts:56]: Deleting the instance key could be problematic, as iirc cards a re detached then reattached to move. Instead, you could add a hook to remove it at the end of the player's next turn.

File: common/cards/hermits/shubbleyt-rare.ts [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+					if (!modalResult) return 'SUCCESS'
+					if (!modalResult.result) return 'SUCCESS'
+
+					const topCard = player.pile.shift()
+					if (!topCard) return 'SUCCESS'
+					player.pile.push(topCard)
+
+					return 'SUCCESS'
+				},
+				onTimeout() {
```
Comment (Tyrannicodin) [inline common/cards/hermits/shubbleyt-rare.ts:67]: Again, empty timeout

File: common/cards/hermits/smajor1995.ts [hunk 9]
```diff
@@ -57,6 +57,9 @@ class Smajor1995RareHermitCard extends HermitCard {
 
 					return 'SUCCESS'
 				},
+				onTimeout() {
```
Comment (Tyrannicodin) [inline common/cards/hermits/smajor1995.ts:60]: Empty timeout

File: common/cards/hermits/smajor1995.ts [hunk 10]
```diff
@@ -26,7 +26,7 @@ class Smajor1995RareHermitCard extends HermitCard {
 				name: 'To Dye For',
 				cost: ['any', 'any', 'any'],
 				damage: 70,
-				power: 'After your attack, select one of your Hermits. Items attached to this Hermit become any type.',
+				power: 'Select one of your Hermits. Items attached to this Hermit become any type.',
```
Comment (Tyrannicodin) [inline common/cards/hermits/smajor1995.ts]: This was added specifically due to sense's comment

File: common/cards/hermits/solidaritygaming-rare.ts [hunk 11]
```diff
@@ -5,7 +5,7 @@ import {RowPos} from '../../types/cards'
 import {RowStateWithHermit} from '../../types/game-state'
 import {getNonEmptyRows} from '../../utils/board'
 import HermitCard from '../base/hermit-card'
-import {applyAilment, removeAilment} from '../../utils/board'
+import { applyAilment, removeAilment } from '../../utils/board'
```
Comment (Tyrannicodin) [inline common/cards/hermits/solidaritygaming-rare.ts]: Potentially formatting here

File: common/cards/hermits/solidaritygaming-rare.ts [hunk 12]
```diff
@@ -40,17 +39,15 @@ class SolidaritygamingRareHermitCard extends HermitCard {
...
 				const ailmentsToRemove = game.state.ailments.filter((ail) => {
-					return (
-						ail.targetInstance === row.hermitCard.cardInstance && ail.ailmentId === 'protection'
-					)
+					return ail.targetInstance === row.hermitCard.cardInstance && ail.ailmentId === "protection"
 				})
-
-				ailmentsToRemove.forEach((ail) => {
+				
+				ailmentsToRemove.map((ail) => {
```
Comment (Tyrannicodin) [inline common/cards/hermits/solidaritygaming-rare.ts]: This should be forEach

File: common/cards/hermits/solidaritygaming-rare.ts [hunk 13]
```diff
@@ -40,17 +39,15 @@ class SolidaritygamingRareHermitCard extends HermitCard {
...
-
+			
 			player.board.rows.forEach((row) => {
 				if (!row.hermitCard) return
 
 				const ailmentsToRemove = game.state.ailments.filter((ail) => {
-					return (
-						ail.targetInstance === row.hermitCard.cardInstance && ail.ailmentId === 'protection'
-					)
+					return ail.targetInstance === row.hermitCard.cardInstance && ail.ailmentId === "protection"
```
Comment (Tyrannicodin) [inline common/cards/hermits/solidaritygaming-rare.ts]: Formatting? Not sure

File: common/cards/items/index.ts [hunk 14]
```diff
@@ -18,7 +18,7 @@ import TerraformCommonItemCard from './terraform-common'
 import TerraformRareItemCard from './terraform-rare'
 import MinerCommonItemCard from './miner-common'
 import MinerRareItemCard from './miner-rare'
-import ClankCommonItemCard from './coin-common'
+import ClankCommonItemCard from './clank-common'
```
Comment (Tyrannicodin) [inline common/cards/items/index.ts]: This should be coins, but it doesn't matter as we're changing it soon

File: common/cards/single-use/brush.ts [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+		return true
+	}
+
+	override onAttach(game: GameModel, instance: string, pos: CardPosModel) {
+		const {player} = pos
+
+		player.hooks.onApply.add(instance, () => {
+			game.addModalRequest({
+				playerId: player.id,
+				data: {modalId: 'selectCards', payload: {
```
Comment (Tyrannicodin) [inline common/cards/single-use/brush.ts]: Formatting here

File: common/cards/single-use/chest.ts [hunk 16]
```diff
@@ -30,21 +30,30 @@ class ChestSingleUseCard extends SingleUseCard {
 
 		game.addModalRequest({
 			playerId: player.id,
-			data: {modalId: this.id},
+			data: {modalId: 'selectCards', payload: {
```
Comment (Tyrannicodin) [inline common/cards/single-use/chest.ts]: Formatting

File: common/cards/single-use/glowstone.ts [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+					const cards: Array<CardT> = modalResult.cards
+
+					const selectedCards = opponentPlayer.pile.slice(0,3).filter((c) => c !== cards[0])
+					opponentPlayer.pile.filter((c) => !selectedCards.includes(c))
+					selectedCards.map((c) => opponentPlayer.pile.unshift(c))
+					opponentPlayer.hand.push(cards[0])
+
+					return 'SUCCESS'
+				},
+				onTimeout() {},
```
Comment (Tyrannicodin) [inline common/cards/single-use/glowstone.ts]: Empty timeout

File: common/cards/single-use/brush.ts [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+					if (!modalResult) return 'FAILURE_INVALID_DATA'
+
+					const cards: Array<CardT> = modalResult.cards
+
+					player.pile.filter((c) => !cards.includes(c))
+					cards.reverse().map((c) => player.pile.unshift(c))
+
+					return 'SUCCESS'
+				},
+				onTimeout() {},
```
Comment (Tyrannicodin) [inline common/cards/single-use/brush.ts]: Empty timeout

File: common/cards/single-use/lantern.ts [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+							primaryButton: {
+								text: "Close",
+								variant: "default"
+							},
+						}},
+						onResult() {
+							return 'SUCCESS'
+						},
+						onTimeout() {
+							return
```
Comment (Tyrannicodin) [inline common/cards/single-use/lantern.ts]: empty timeout

File: common/cards/single-use/lantern.ts [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+				}},
+				onResult(modalResult) {
+					if (!modalResult) return 'FAILURE_INVALID_DATA'
+
+					player.pile.filter((c) => !modalResult.cards.includes(c))
+					modalResult.cards.reverse().map((c: CardT) => player.hand.push(c))
+
+					game.addModalRequest({
+						playerId: opponentPlayer.id,
+						data: {modalId: 'selectCards', payload: {
```
Comment (Tyrannicodin) [inline common/cards/single-use/lantern.ts]: Formatting

File: common/cards/single-use/lantern.ts [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+							return 'SUCCESS'
+						},
+						onTimeout() {
+							return
+						},
+					})
+
+					return 'SUCCESS'
+				},
+				onTimeout() {},
```
Comment (Tyrannicodin) [inline common/cards/single-use/lantern.ts]: Empty timeout

**Review by Tyrannicodin — CHANGES_REQUESTED** [review]

File: common/ailments/museum-collection.ts (hunk 2 above)
Comment (alicetouhou) [reply common/ailments/museum-collection.ts:1]: how so?

File: common/cards/hermits/bigbst4tz2-rare.ts (hunk 5 above)
Comment (alicetouhou) [reply common/cards/hermits/bigbst4tz2-rare.ts:61]: you're right i will rewrite this

File: common/cards/hermits/pixlriffs-rare.ts (hunk 7 above)
Comment (alicetouhou) [reply common/cards/hermits/pixlriffs-rare.ts:56]: i checked this and it worked, so that would be surprising

**Review by Tyrannicodin — APPROVED** [review]
